### PR TITLE
text about EAP-TLS use of client side certificates

### LINKED ISSUE #10: architecture: EAP-TLS with RADIUS
Empty section now. Any ideas?

### PR DESCRIPTION
**Pull request by mcr**
close #10

### CODE REVIEW

**Comment by mcr** [comment]
@alandekok could maybe review.

**Comment by alandekok** [comment]
RFC 3748 says:

```
       EAP methods can assume a minimum EAP MTU of 1020 octets in the
       absence of other information.
```

EAP-TLS also requires exchange of the full certificate chain (up to the CA).   From RFC 5216:

```
   In contrast to the EAP-TLS server, the EAP-TLS peer may not have
   Internet connectivity.  Therefore, the EAP-TLS server SHOULD provide
   its entire certificate chain minus the root to facilitate certificate
   validation by the peer.
```

And then for the client cert:

```
  The certificate message contains a public key certificate chain for
   either a key exchange public key (such as an RSA or Diffie-Hellman
   key exchange public key) or a signature public key (such as an RSA or
   Digital Signature Standard (DSS) signature public key). 
```

In practice, this always happens.

This means that if (a) certificates are big, and (b) there are lots of them, it's possible to have ~50 round trips.  Which is annoying.

There has been discussion about fixing this, but no progress.

**Comment by mcr** [comment]
> RFC 3748 says:
> 
> ```
>        EAP methods can assume a minimum EAP MTU of 1020 octets in the
>        absence of other information.
> ```

That's the minimum, but which I guess leads to a maximum number of segments to send a 2-kilobyte certificate.

> And then for the client cert:
> 
> ```
>   The certificate message contains a public key certificate chain for
>    either a key exchange public key (such as an RSA or Diffie-Hellman
>    key exchange public key) or a signature public key (such as an RSA or
>    Digital Signature Standard (DSS) signature public key). 
> ```
> 
> In practice, this always happens.

sorry, what is "this" in this context?

> 
> This means that if (a) certificates are big, and (b) there are lots of them, it's possible to have ~50 round trips. Which is annoying.
> 
> There has been discussion about fixing this, but no progress.

DANCE can't eliminate the server certificates, but we think we can eliminate the client certificate.

File: draft-ietf-dance-architecture.md [hunk 1]
```diff
@@ -236,11 +236,25 @@ Issue #9
...
 
-Issue #10
+Enterprise EAP methods use a version of TLS to form a secure transport.
+Client and server-side certificates are used as credentials.
+EAP-TLS does not run over TCP, but rather over a reliable transport provided by EAP.
+To keep it simple the EAP "window" is always one, and there are various amounts of overhead that needs to be accounted for, and the EAP segment size is often noticeably smaller than the normal ethernet 1500 bytes.
+{{?RFC3748}} does guarantee a minimum payload of 1020 bytes.
+
+The client side certificates are often larger than 1500 bytes and can take two or three round trip times to transport from the supplicant to the authenticator.
+In worst case scenarios, which are common with EDUROAM {{?RFC7593}}, the EAP packets are transported some distance, easily across the entire planet, to an authenticator that likely has actually been the entity that issued the certificate.
```
Comment (mcr) [inline draft-ietf-dance-architecture.md:254]:
```suggestion
In worst case scenarios, which are common with EDUROAM {{?RFC7593}}, the EAP packets are transported some distance, easily across the entire planet.
The authenticating system (the "authentication server" in EAP terms) is a system at the institute that issued the client side certificate, and so already has access to the entire client certificate.
Transferring the client certificate is redundant.
```

**Review by oej — APPROVED** [review]
Looks good

**Comment by alandekok** [comment]
NIT: eduroam is always lowercase.  Why?  I dunno...

**Pull request merged**